runtime/src/precompiles/metagraph.rs

```diff
@@ -0,0 +1,381 @@
+extern crate alloc;
+use crate::precompiles::{get_method_id, get_slice};
+use crate::Runtime;
+use fp_evm::{
+    ExitError, ExitSucceed, PrecompileFailure, PrecompileHandle, PrecompileOutput, PrecompileResult,
+};
+use sp_core::{ByteArray, U256};
+use sp_std::vec;
+pub const METAGRAPH_PRECOMPILE_INDEX: u64 = 2050;
+pub struct MetagraphPrecompile;
+
+const NO_HOTKEY: &str = "no hotkey";
+
+impl MetagraphPrecompile {
+    pub fn execute(handle: &mut impl PrecompileHandle) -> PrecompileResult {
+        let txdata = handle.input();
+        let method_id = get_slice(txdata, 0, 4)?;
+        let method_input = txdata
+            .get(4..)
+            .map_or_else(vec::Vec::new, |slice| slice.to_vec()); // Avoiding borrowing conflicts
+
+        match method_id {
+            id if id == get_method_id("getUidCount(uint16)") => Self::get_uid_count(&method_input),
+            id if id == get_method_id("getIsSubnetRegistered(uint16)") => {
```

Comment:
Move this method to subnets precompile. It will be implemented in the issue #1067